File: py/009 Palindrome Number/PalindromeNumber.py

# 本题题意：判斷回文數字

class Solution(object):
    def isPalindrome(self, x):
        """
        :type x: int
        :rtype: bool
        """
        s = str(x)
        length = len(s)
        for i in range(int(length / 2)):
            if s[i] != s[length - 1 - i]:
                return False
        return True

File: swift/009 Palindrome Number/PalindromeNumber.swift

// 本题题意：判斷回文數字
class Solution {
    func isPalindrome(_ x: Int) -> Bool {
        let str = String(x), length = str.characters.count
        for i in 0...(length / 2) {
            if str[str.index(str.startIndex, offsetBy: i)] != str[str.index(str.startIndex, offsetBy: (length - 1 - i))] {
                return false
            }
        }
        return true
    }
}
